### ISSUE TITLE
ConnectionNotFound or bad routing results close to the border of the map.

### ISSUE THREAD

**Issue by easbar** [graphhopper/graphhopper#1610]
I ran into some cases where the edge-based routing did not find a path close to the 'border' of the graph.

For example this: http://localhost:8989/maps/?point=48.250955%2C13.033712&point=48.252921%2C13.023304&locale=en-US&vehicle=car&weighting=fastest&elevation=false&edge_based=true&use_miles=false&layer=Omniscale
does not find a connection on `bayern-140101.osm.pbf` (for different algos dijkstra/astar ch/non-ch does not matter). With edge_based=false a route is calculated, but it is rather useless: 

![image](https://user-images.githubusercontent.com/17603532/57280798-d5c8c780-70a9-11e9-99d9-5bb6977038e6.png)

Not sure if anything can or needs to be done about this. The only thing that came to my mind is that we do a connectivity check in `PrepareRoutingSubnetworks`, but it does not pay attention to turn restrictions.

**Comment by karussell** [comment]
Oh, this is indeed tricky and as you say: to fix this we would have to do the PrepareRoutingSubnetworks with considering the turn restrictions. Not that this is not possible but maybe the benefit is too low to justify the slower preparation?

**Comment by easbar** [comment]
> Not that this is not possible but maybe the benefit is too low to justify the slower preparation?

Yes I agree :) Especially because then we would get a bad route similar to the one in the screenshot instead of ConnectionNotFound. I was thinking if there is a way to detect/warn if a from/to point is (too) close to the border, but no idea really.

**Comment by easbar** [comment]
I mean 'bad route' depends on the way of thinking. Assuming that the map in use is *the* graph we are routing on it is not a bad route, but still can be somewhat unexpected if one does not have in mind that the graph 'ends' somewhere.

**Comment by karussell** [comment]
If the roads would be not reachable with turn restrictions we make them inaccessible in PrepareRoutingSubnetworks, so this would result in a different snapping but the route would probably be ok (as snapping could be 1km away)

**Comment by easbar** [comment]
> can be somewhat unexpected if one does not have in mind that the graph 'ends' somewhere.

would it be feasible to draw the polygon that represents the border of the graph (should be similar to how the polygons are obtained in isochrones) ? could be useful at least for debug view (#1572). the bounding box can be misleading with this regard.

**Comment by karussell** [comment]
> would it be feasible to draw the polygon that represents the border of the graph (should be similar to how the polygons are obtained in isochrones) ?

The algorithm of the isochrones can't be used as it currently does not scale to large areas. But #1572 would solve the issue already without such a polygon (?)

**Comment by easbar** [comment]
>  But #1572 would solve the issue already without such a polygon (?)

Oh yes ;) sure if we see the graph no need to draw a border :D
